src/utils/dockerDirManager.ts

```diff
@@ -0,0 +1,303 @@
+
+    try {
+      if ((await fs.promises.stat(currentSocketPath)).isSocket()) {
+        return currentContext;
+      }
+      console.log(`Invalid existing context "${ currentContext }": ${ currentSocketUri } is not a socket; overriding context.`);
+    } catch (ex) {
+      console.log(`Could not read existing docker socket ${ currentSocketUri }, overriding context "${ currentContext }": ${ ex }`);
+    }
+
+    return this.contextName;
+  }
+
+  /**
+   * Determines whether the passed credential helper is working.
+   * @param helperName The cred helper name, without the "docker-credential-" prefix.
+   */
+  credHelperWorking(helperName: string): Promise<boolean> {
+    const helperBin = `docker-credential-${ helperName }`;
+    const logMsg = `Credential helper "${ helperBin }" is not functional`;
+    let proc: any;
+
+    try {
+      proc = spawn(helperBin, ['list']);
+    } catch {
+      console.log(logMsg);
+
+      return Promise.resolve(false);
+    }
+
+    return new Promise( (resolve) => {
+      proc.on('exit', (code: number) => {
+        if (code) {
+          console.log(logMsg);
+          resolve(false);
+        }
+        resolve(true);
+      });
+    });
+  }
+
+  /**
+   * Returns the default cred helper name for the current platform.
+   */
+  getDefaultDockerCredsStore(): string {
+    const platform = os.platform();
+
+    if (platform.startsWith('win')) {
+      return 'wincred';
+    } else if (platform === 'darwin') {
+      return 'osxkeychain';
+    } else if (platform === 'linux') {
+      return 'secretservice';
+    } else {
+      throw new Error(`platform "${ platform }" is not supported`);
+    }
+  }
+
+  /**
+   * Ensures that the rancher-desktop docker context exists.
+   * @param socketPath Path to the rancher-desktop specific docker socket.
+   * @param kubernetesEndpoint Path to rancher-desktop Kubernetes endpoint.
+   */
+  async ensureDockerContext(socketPath: string, kubernetesEndpoint?: string): Promise<void> {
+    const contextContents = {
+      Name:      this.contextName,
+      Metadata:  { Description: 'Rancher Desktop moby context' },
+      Endpoints: {
+        docker: {
+          Host:          `unix://${ socketPath }`,
+          SkipTLSVerify: false,
+        },
+      } as Record<string, {Host: string, SkipTLSVerify: boolean, DefaultNamespace?: string}>,
+    };
+
+    if (kubernetesEndpoint) {
+      contextContents.Endpoints.kubernetes = {
+        Host:             kubernetesEndpoint,
+        SkipTLSVerify:    true,
+        DefaultNamespace: 'default',
+      };
+    }
+
+    console.debug(`Updating docker context: writing to ${ this.dockerContextPath }`, contextContents);
+
+    await fs.promises.mkdir(path.dirname(this.dockerContextPath), { recursive: true });
+    await fs.promises.writeFile(this.dockerContextPath, JSON.stringify(contextContents));
+  }
+
+  /**
+   * Clear the docker context; this is used for factory reset.
```

Comment:
This is also used in WSL when we're using containerd. But not in Lima, for some reason.

Reply:
Yes, I forgot to call it in `LimaBackend`. I've changed it so that it is only called in each backend's `factoryReset()`. This should serve our purposes.

Reply:
The comment should be updated.

Reply:
Sorry could you provide more info? I'm not sure how the comment would be updated. Isn't this method used only in factory resets? And it only clears the docker context? I'm confused

Reply:
It's used in factory reset _and_ on containerd-based startups, not _just_ factory resets.